pkgs/development/python-modules/mtcnn/default.nix

```diff
@@ -2,50 +2,57 @@
   lib,
   buildPythonPackage,
   fetchFromGitHub,
-  opencv-python,
-  pytestCheckHook,
-  pythonOlder,
+
+  # build-system
   setuptools,
+
+  # dependencies
+  joblib,
+  keras,
+  lz4,
   tensorflow,
+  pythonAtLeast,
+  distutils,
+
+  # tests
+  pytestCheckHook,
 }:
 
-buildPythonPackage {
+buildPythonPackage rec {
   pname = "mtcnn";
-  version = "0.1.1";
+  version = "1.0.0";
   pyproject = true;
 
-  disabled = pythonOlder "3.7";
-
   src = fetchFromGitHub {
     owner = "ipazc";
     repo = "mtcnn";
-    # No tags / releases; using commit: https://github.com/ipazc/mtcnn/commit/3208d443a8f01d317c65d7c97a03bc0a6143c41d
-    rev = "3208d443a8f01d317c65d7c97a03bc0a6143c41d";
-    hash = "sha256-GXUrLJ5XD6V2hT/gjyYSuh/CMMw2xIXKBsYFvQmbLYs=";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-gp+jfa1arD3PpJpuRFKIUznV0Lyjt3DPn/HHUviDXhk=";
   };
 
-  postPatch = ''
-    substituteInPlace setup.py \
-      --replace-fail "setup, setuptools" "setup, find_packages"\
-      --replace-fail "setuptools.find_packages" "find_packages"\
-      --replace-fail "keras>=2.0.0" ""\
-      --replace-fail "tests_require=['nose']," ""
-  '';
-
   build-system = [ setuptools ];
 
-  dependencies = [
-    opencv-python
-    tensorflow
-  ];
+  dependencies =
+    [
+      joblib
+      lz4
+      tensorflow
```

Comment:
tensorflow seems to be an optional dependency
https://github.com/ipazc/mtcnn/blob/117e9c6d1a4a740ac497f65026161658d28dab1e/setup.py#L41-L48

Reply:
Indeed, it seems to work fine without it.